### PR TITLE
feat!: transfer for multiple addresses

### PR DESCRIPTION
**Pull request by Torres-ssf**
- Close: https://github.com/FuelLabs/fuels-ts/issues/161

## Breaking Changes:
- [Changed](https://github.com/FuelLabs/fuels-ts/pull/2333/commits/e0e332bf1cb20d8ad8f211dbdbce15a7967f77a6#diff-3409481ab69cbec51784c01628082462a0c87c332fd2e0cdf97fb93d9a59b2fdR314-R315) `BaseInvocationScope.addTransfer` parameters

### CODE REVIEW

File: apps/docs/src/guide/wallets/wallet-transferring.md [hunk 1]
```diff
@@ -18,6 +18,12 @@ When transferring the base chain coin like ETH, you can omit the `assetId`:
 
 <<< @/../../docs-snippets/src/guide/wallets/wallet-transferring.test.ts#wallet-transferring-3{ts:line-numbers}
 
+## Transferring To Multiple Wallets
+
+To transfer assets to multiple wallets, use the `Account.multiTransfer` method:
```
Comment (maschad) [inline apps/docs/src/guide/wallets/wallet-transferring.md:23]: We may want to call the method `batchTransfer` as that's the more familiar [parlance in the space](https://coinguides.org/batching-bitcoin-transactions/)

**Review by maschad — DISMISSED** [review]
Awesome job on this 🚀  -  just left one suggestion

File: apps/docs/src/guide/wallets/wallet-transferring.md (hunk 1 above)
Comment (Torres-ssf) [reply apps/docs/src/guide/wallets/wallet-transferring.md:23]: @maschad That is a good suggestion, I agree that `batchTransfer` makes it clear that the method is designed to handle multiple transfers in a single transaction

**Review by danielbate — COMMENTED** [review]
This is a great feature and adds a lot of value. My only questions is around the spec in #161.

> Also this should be integrated to enable queue txs instead of single tx

I can see we are using a single tx here, should we be queueing?

**Comment by fuel-service-user** [comment]
✨ A PR has been created under the `rc-2333` tag on `fuels-wallet` repo.
https://github.com/FuelLabs/fuels-wallet/pull/1311

File: packages/account/src/account.ts [hunk 2]
```diff
@@ -417,17 +409,66 @@ export class Account extends AbstractAccount {
...
+   * @param destination - The destination address for the transfer.
+   * @param amount - The amount to transfer.
+   * @param assetId - (Optional) The asset ID for the transfer. If not provided, the base asset ID from the provider will be used.
+   * @returns The updated transaction request with the added transfer.
+   */
+  addTransfer(
+    request: ScriptTransactionRequest,
+    destination: string | AbstractAddress,
+    amount: BigNumberish,
+    assetId?: BytesLike
```
[the remark this answers is not in the document]
Comment (Torres-ssf) [reply packages/account/src/account.ts]: https://github.com/FuelLabs/fuels-ts/pull/2333/commits/31a79facf437d3f1bd4cc6eb1a1a1be0cb08c37d

File: packages/account/src/account.ts [hunk 3]
```diff
@@ -602,6 +643,33 @@ export class Account extends AbstractAccount {
     return this.provider.simulate(transactionRequest, { estimateTxDependencies: false });
   }
 
+  private validateTransferAmount(amount: BigNumberish) {
```
[the remark this answers is not in the document]
Comment (Torres-ssf) [reply packages/account/src/account.ts:646]: https://github.com/FuelLabs/fuels-ts/pull/2333/commits/5bf92d0b652dd3cafff61757a322e18d38e4d48a

File: packages/account/src/account.ts [hunk 4]
```diff
@@ -602,6 +643,33 @@ export class Account extends AbstractAccount {
...
+    if (bn(amount).lte(0)) {
+      throw new FuelError(
+        ErrorCode.INVALID_TRANSFER_AMOUNT,
+        'Transfer amount must be a positive number.'
+      );
+    }
+  }
+
+  private async estimateAndFundTransaction(
```
[the remark this answers is not in the document]
Comment (Torres-ssf) [reply packages/account/src/account.ts:656]: https://github.com/FuelLabs/fuels-ts/pull/2333/commits/5bf92d0b652dd3cafff61757a322e18d38e4d48a

**Comment by Torres-ssf** [comment]
> This is a great feature and adds a lot of value. My only questions is around the spec in #161.
> 
> > Also this should be integrated to enable queue txs instead of single tx
> 
> I can see we are using a single tx here, should we be queueing?

@danielbate I reckon you were mentioning something like this: https://github.com/FuelLabs/fuels-ts/pull/2333/commits/a56d59b9c5397c8831e393be5acfc1ff9249ccde, right?

File: apps/docs/src/guide/wallets/wallet-transferring.md (hunk 1 above)
Comment (Torres-ssf) [reply apps/docs/src/guide/wallets/wallet-transferring.md]: https://github.com/FuelLabs/fuels-ts/pull/2333/commits/b06df0153361f0de37c1d151ba0767a17f9fb9af

**Review by maschad — APPROVED** [review]

**Review by Dhaiwat10 — APPROVED** [review]
Great work 🚀

**Comment by danielbate** [comment]
> @danielbate I reckon you were mentioning something like this: https://github.com/FuelLabs/fuels-ts/commit/a56d59b9c5397c8831e393be5acfc1ff9249ccde, right?

But this still uses a single tx? A single transaction request with multiple recipient outputs. Whereas this issue also asks for:

> queuing of transactions (i.e. once X tx is done, do X + 1, etc)

**Comment by Torres-ssf** [comment]
> But this still uses a single tx? A single transaction request with multiple recipient outputs. Whereas this issue also asks for:
> 
> > queuing of transactions (i.e. once X tx is done, do X + 1, etc)

@danielbate Thanks for your explanation. I see your point. 

I've created a specific [issue](https://github.com/FuelLabs/fuels-ts/issues/2361) for this since it's a completely different thing from multiple addresses transferring.

**Review by danielbate — APPROVED** [review]

**Pull request merged**